test/test_gatherv.f90

```diff
@@ -0,0 +1,111 @@
+program test_gatherv
+  use libmpifx_module
+  implicit none
+
+  type(mpifx_comm) :: mycomm
+  real, allocatable :: send1(:), send2(:,:)
+  real, allocatable :: recv1(:), recv2(:,:)
+  real :: send0
+  integer, allocatable :: recvcounts(:)
+  integer, allocatable :: displs(:)
+  integer :: ii, nrecv  
+  character(100) :: formstr
+  character(*), parameter :: label = "(I2.2,'-',I3.3,'|',1X"
+
+  call mpifx_init()
+  call mycomm%init()
+
+  ! I1 -> I1
+  if (mycomm%master) write(*, *) 'Test gather rank=1 -> rank=1'
+
+  allocate(send1(mycomm%rank+1)) 
+  send1 = 1.0*(mycomm%rank+1)
+  if (mycomm%master) then
+    ! recv1 size is 1+2+3+...+mycomm%size 
+    nrecv = mycomm%size*(mycomm%size+1)/2
+    allocate(recv1(nrecv))
+    allocate(recvcounts(mycomm%size))
+    do ii = 1, mycomm%size  
+      recvcounts(ii) = ii 
+    end do 
+  else
+    allocate(recv1(0))
+    allocate(recvcounts(0))
+  end if
+
+  write(*, *) 'id:',mycomm%rank, "Send1 buffer:", send1(:)
+
+  call mpifx_gatherv(mycomm, send1, recv1, recvcounts)
+  
+  call mpifx_barrier(mycomm)
```

Comment:
Are the barrier statements needed here? The gatherv call is blocking (https://stackoverflow.com/questions/9284419/is-mpi-reduce-blocking-or-a-natural-barrier) and at least on a shared memory machine this example isn't fast enough to cause problems (test_gather doesn't use these).